Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/LuceneIndexController.java

```diff
@@ -0,0 +1,97 @@
+package edu.asu.conceptpower.servlet.web;
+
+import java.security.Principal;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import edu.asu.conceptpower.servlet.core.IIndexService;
+import edu.asu.conceptpower.servlet.core.IndexingEvent;
+import edu.asu.conceptpower.servlet.exceptions.LuceneException;
+import edu.asu.conceptpower.servlet.lucene.ILuceneDAO;
+
+/**
+ * This class provides methods for deleting and viewing lucene indexes
+ * 
+ * @author mkarthik90
+ *
+ */
+@Controller
+public class LuceneIndexController {
+
+    @Autowired
+    private IIndexService manager;
+
+    @Autowired
+    @Qualifier("luceneDAO")
+    private ILuceneDAO dao;
+
+    /**
+     * Retrives latest indexing time and number of concepts indexed so far
+     * 
+     * @param model
+     * @return
+     */
+    @RequestMapping(value = "auth/luceneIndex", method = RequestMethod.GET)
+    public String showLuceneIndex(ModelMap model) {
+        IndexingEvent bean = dao.getTotalNumberOfWordsIndexed();
+        model.addAttribute("bean", bean);
+        return "/auth/luceneIndex";
+    }
+
+    /**
+     * This method deletes all the index in Lucene and recreates all the index
+     * from Wordnet and from CCP db40 database
+     * 
+     * @param req
+     * @param principal
+     * @param model
+     * @return
+     */
+    @RequestMapping(value = "auth/indexLuceneWordNet", method = RequestMethod.POST, produces = "application/json")
+    public synchronized @ResponseBody IndexingEvent indexConcepts(HttpServletRequest req, Principal principal, ModelMap model) {
```

Comment:
I wouldn't do this, this would create a bottle neck (all threads have to wait for the others to finish). It probably wouldn't be a big deal here (given that it's unlikely many people would start the indexer at the same time), but it's bad design and creates redundant calculations. Instead use a flag to check if the indexer is already running (e.g. AtomicBoolean) and only continue if it's not running. Otherwise tell the user that the indexer is currently running.